apps/web-giddh/src/app/shared/primary-sidebar/company-branch/company-branch.component.ts

```diff
@@ -0,0 +1,429 @@
+        this.isCompanyRefreshInProcess$ = this.store.pipe(select(state => state.session.isRefreshing), takeUntil(this.destroyed$));
+        this.currentCompanyBranches$ = this.store.pipe(select(appStore => appStore.settings.branches), takeUntil(this.destroyed$));
+
+        this.store.pipe(select((state: AppState) => state.session.companies), takeUntil(this.destroyed$)).subscribe(companies => {
+            if (!companies || companies.length === 0) {
+                return;
+            }
+
+            let orderedCompanies = orderBy(companies, 'name');
+            this.companyList = orderedCompanies;
+            this.companyListForFilter = orderedCompanies;
+            this.companies$ = observableOf(orderedCompanies);
+
+            if (this.searchCompany) {
+                this.filterCompanyList(this.searchCompany);
+            }
+        });
+
+        this.store.pipe(select(state => state.session.activeCompany), takeUntil(this.destroyed$)).subscribe(selectedCmp => {
+            if (selectedCmp && selectedCmp?.uniqueName === this.generalService.companyUniqueName) {
+                this.selectedCompany = observableOf(selectedCmp);
+                this.activeCompany = selectedCmp;
+                this.companyInitials = this.generalService.getInitialsFromString(selectedCmp.name);
+                this.activeDesign = (this.activeCompany?.name.charCodeAt(0) <= 77) ? 1 : 2;
+
+                if(!this.companyBranches?.branches) {
+                    this.companyBranches = selectedCmp;
+                }
+
+                this.currentCompanyBranches$.subscribe(response => {
+                    if (response && response.length) {
+                        this.branchList = response.sort(this.generalService.sortBranches);
+                        this.currentCompanyBranches = this.branchList;
+                        this.companyBranches.branches = this.branchList;
+                        this.companyBranches.branchCount = this.branchList?.length
+                        this.changeDetectorRef.detectChanges();
+                    }
+                });
+            }
+        });
+
+        this.store.pipe(select(appStore => appStore.session.currentOrganizationDetails), takeUntil(this.destroyed$)).subscribe((organization: Organization) => {
+            this.currentBranchUniqueName = "";
+
+            if (organization && organization.details && organization.details.branchDetails) {
+                this.currentBranchUniqueName = organization.details.branchDetails.uniqueName;
+            }
+        });
+    }
+
+    /**
+     * Releases the occupied memory
+     *
+     * @memberof CompanyBranchComponent
+     */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+    }
+
+    /**
+     * Company filtering method
+     *
+     * @param {*} ev Modal change event
+     * @memberof CompanyBranchComponent
+     */
+    public filterCompanyList(event: any): void {
```

Comment:
Please check this method is also present in `primary-sidebar.component.ts`, do remove it from there if unused.